vllm/spec_decode/proposer_worker_base.py

```diff
@@ -0,0 +1,41 @@
+from abc import ABC, abstractmethod
+from typing import List, Optional, Tuple
+
+from vllm.sequence import ExecuteModelRequest, SamplerOutput
+from vllm.spec_decode.interfaces import SpeculativeProposer
+from vllm.worker.worker_base import WorkerBase
+
+
+class ProposerWorkerBase(WorkerBase, SpeculativeProposer):
+    """Interface for proposer workers"""
+
+    @abstractmethod
+    def sampler_output(
+        self,
+        execute_model_req: ExecuteModelRequest,
+        sample_len: int,
+    ) -> Tuple[Optional[List[SamplerOutput]], bool]:
+        raise NotImplementedError
+
+    def set_include_gpu_probs_tensor(self):
+        """Implementation optional"""
+        pass
+
+
+class NonLLMProposerWorkerBase(ProposerWorkerBase, ABC):
+    """Proposer worker which does not use a model with kvcache"""
+
+    def execute_model(
+            self,
+            execute_model_req: Optional[ExecuteModelRequest] = None) -> None:
+        pass
+
+    def determine_num_available_blocks(self) -> None:
+        pass
```

Comment:
The typing / return type is wrong (should be Tuple[int, int]).

Hm.. maybe we just raise NotImplementedError with a comment? We don't ever run profiling on the proposal method (always target model as it's assumed to have larger KV).

Reply:
Ah of course sorry, I just copied these no-op methods from `NGramWorker`.